File: Modulo 1/Clase02/app/src/main/java/com/area51/clase02/MainActivity.java
package com.area51.clase02;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    //Declaramos las variables que se van a vincular
    //a los componentes
    private EditText nombre, apellido;
    private Button obtener, enviar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Vincular las variables a los componentes xml
        nombre = findViewById(R.id.etNombre);
        apellido = findViewById(R.id.etApellido);
        obtener = findViewById(R.id.btnObtener);
        enviar = findViewById(R.id.btnEnviar);
    }

    @Override
    protected void onResume() {
        super.onResume();

        //Creamos el evento onClick al boton obtener
        obtener.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                //Obtenemos los datos de los edittext
                String valorNombre = nombre.getText().toString();
                String valorApellido = apellido.getText().toString();

                //Mostramos el mensaje flotante con el texto ingresado
                Toast.makeText(
                        MainActivity.this,
                        "Nombre: " + valorNombre +
                                " - Apellido: " + valorApellido,
                        Toast.LENGTH_SHORT)
                        .show();

            }
        });
        enviar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String valorNombre = nombre.getText().toString();
                String valorApellido = apellido.getText().toString();

                //Pasar a otro activity
                Intent intent = new Intent(
                        MainActivity.this,
                        PantallaDosActivity.class
                );
                intent.putExtra("v_nombre", valorNombre);
                intent.putExtra("v_apellido", valorApellido);
                startActivity(intent);

            }
        });
    }
}

File: Modulo 3/ClaseKotlin/app/src/main/java/pe/area51/clasekotlin/MainActivity.kt
package pe.area51.clasekotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

//public class MaiActivity extends AppCompatActivity{}

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onResume() {
        super.onResume()

        btnMostrar.setOnClickListener {
            val nombre = etNombre.text.toString()
            val genero = spGenero.selectedItem.toString()

            if (nombre.isEmpty()) {
                etNombre.error = "El campo es requerido"
                return@setOnClickListener
            } else
                etNombre.error = null

            Toast.makeText(this,
                    "Nombre $nombre - Genero $genero",
                    Toast.LENGTH_SHORT).show()
        }
        btnOpciones.setOnClickListener {
            val intent = Intent(this, OpcionesActivity::class.java)
            startActivity(intent)
        }
    }

    fun nombre(valor: String): String {
        return ""
    }
}
